### PR TITLE
Web Bluetooth Scanning Sample, initial draft.

### PR DESCRIPTION
**Pull request by scheib**
Introducing the most basic of scanning samples.  Intent is to show
that scanning can be started, demonstrate the permissions UI,
and that advertisement packets are received.

Several details are omitted.  E.g. filtering by service data,
and showing advertisement packet contents.

### CODE REVIEW

**Comment by scheib** [comment]
@dougt, @beaufortfrancois, Please take a look for review.

File: web-bluetooth/scan.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  if (document.querySelector('#allAdvertisements').checked) {
+    options.acceptAllAdvertisements = true;
+  } else {
+    options.filters = filters;
+  }
+
+  try {
+    log('Requesting Bluetooth Scan...');
+    log('with ' + JSON.stringify(options));
+    await navigator.bluetooth.requestLEScan(options);
```
Comment (beaufortfrancois) [inline web-bluetooth/scan.js]: I think basic demo should at least allow users to stop scanning.
How about something like below?
```js
var scan;

async function onStartButtonClick() {
  /* ... */
  scan = await navigator.bluetooth.requestLEScan(options);
  /* ... */
}

async function onStopButtonClick() {
  if (!scan || !scan.active) {
    return;
  }
  scan.stop();
}
`

File: web-bluetooth/scan.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+    log('Scan started...');
```
Comment (beaufortfrancois) [inline web-bluetooth/scan.js]: Shall we show scan options?

**Review by beaufortfrancois — COMMENTED** [review]
Thank you for writing this. I left some comments
I'm happy to write following ones ;)

**Comment by beaufortfrancois** [comment]
@scheib See https://github.com/GoogleChrome/samples/pull/648

By the way, we may want to add a "promise" test as well.

File: web-bluetooth/scan.js (hunk 2 above)
Comment (scheib) [reply web-bluetooth/scan.js]: Done

File: web-bluetooth/scan.js (hunk 1 above)
Comment (scheib) [reply web-bluetooth/scan.js]: I'd like to keep the code as simple as possible, I've used a simple timer to stop the scan after 10 seconds.

File: web-bluetooth/scan.js (hunk 2 above)
Comment (scheib) [reply web-bluetooth/scan.js]: done

**Comment by scheib** [comment]
> @scheib See #648
> 
> By the way, we may want to add a "promise" test as well.

I suggest we no longer create pairs, since it's been so many years and async-await is now pervasive https://caniuse.com/#feat=async-functions.  Especially if we're creating a new batch of scanning related samples, I don't see the need.

File: web-bluetooth/scan.html [hunk 3]
```diff
@@ -0,0 +1,39 @@
+---
+feature_name: Web Bluetooth / Scanning
+chrome_version: 79
```
Comment (beaufortfrancois) [inline web-bluetooth/scan.html:3]: Let's make sure to update this if feature gets postponed.

File: web-bluetooth/scan.js [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    const scan = await navigator.bluetooth.requestLEScan(options);
+
+    log('Scan started with:');
+    log(' acceptAllAdvertisements: ' + scan.acceptAllAdvertisements);
+    log(' active: ' + scan.active);
+    log(' keepRepeatedDevices: ' + scan.keepRepeatedDevices);
+    log(' filters: ' + JSON.stringify(scan.filters));
+
+    navigator.bluetooth.addEventListener('advertisementreceived', event => {
+      log("Advertisement received.");
```
Comment (beaufortfrancois) [inline web-bluetooth/scan.js]: Nit: Use simple quotes for consistency.

**Review by beaufortfrancois — APPROVED** [review]

File: web-bluetooth/scan.html (hunk 3 above)
Comment (scheib) [reply web-bluetooth/scan.html:3]: OK, also added note that chrome://flags/#enable-experimental-web-platform-features is required now.

**Comment by beaufortfrancois** [comment]
Sample is now live at https://googlechrome.github.io/samples/web-bluetooth/scan.html

**Comment by beaufortfrancois** [comment]
I've added it to https://www.chromestatus.com/feature/5346724402954240

**Pull request merged**